src/functions/utils.php

```diff
@@ -26,40 +26,77 @@ function tribe_array_merge_recursive( array &$array1, array &$array2 ) {
 	}
 }
 
+
 if ( ! function_exists( 'tribe_register_plugin' ) ) {
 	/**
 	 * Checks if this plugin has permission to run, if not it notifies the admin
 	 *
-	 * @param string $file_path   Full file path to the base plugin file
-	 * @param string $main_class  The Main/base class for this plugin
-	 * @param string $version     The version
-	 * @param array  $classes_req Any Main class files/tribe plugins required for this to run
+	 * @param string $file_path    Full file path to the base plugin file
+	 * @param string $main_class   The Main/base class for this plugin
+	 * @param string $version      The version
+	 * @param array  $classes_req  Any Main class files/tribe plugins required for this to run
+	 * @param array  $dependencies an array of dependencies to check
 	 *
 	 * @return bool Indicates if plugin should continue initialization
 	 */
-	function tribe_register_plugin( $file_path, $main_class, $version, $classes_req = array() ) {
-		$tribe_dependency = Tribe__Dependency::instance();
-		$should_plugin_run = true;
-
-		// Checks to see if the plugins are active
-		if ( ! empty( $classes_req ) && ! $tribe_dependency->has_requisite_plugins( $classes_req ) ) {
-			$should_plugin_run = false;
+	function tribe_register_plugin( $file_path, $main_class, $version, $classes_req = array(), $dependencies = array() ) {
+		$tribe_dependency  = Tribe__Dependency::instance();
 
-			$tribe_plugins = new Tribe__Plugins();
-			$admin_notice  = new Tribe__Admin__Notice__Plugin_Download( $file_path );
+		//add all plugins to registered_plugins
```

Comment:
Let's take this opportunity to relocate the guts of this function into `Tribe__Dependency` in a new method and then this function can be a simple helper with no special logic.